src/App.tsx

```diff
@@ -1,24 +1,162 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import * as todoServise from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+const USER_ID = 11092;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoStatus, setTodoStatus] = useState<Status>(Status.ALL);
+  const [loadingTodoIds, setLoadingTodoIds] = useState([0]);
+
+  useEffect(() => {
+    todoServise.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setIsError(true);
+        setErrorMessage('Unable to load todos');
+      });
+  }, []);
+
+  const addTodo = useCallback(async (title: string) => {
+    try {
+      const newTodo = {
+        title,
+        userId: USER_ID,
+        completed: false,
+      };
+
+      setTempTodo({ ...newTodo, id: 0 });
+
+      const createdTodo = await todoServise.createTodo(newTodo);
+
+      setTodos(currTodos => [...currTodos, createdTodo]);
+    } catch {
+      setIsError(true);
+      setErrorMessage('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+    }
+  }, [USER_ID, todos]);
```

Comment:
The 'todos' dependency is not necessary in the 'addTodo' useCallback hook, since it's not used within the function.